File: languages/go/entity/identity/identity.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type Identity struct {
	Mnemonic   string
	Key        string
	PrivateKey string
}

func NewIdentityFromProto(s *proto.Identity) Identity {
	if s == nil {
		return Identity{}
	}
	return Identity{
		Mnemonic:   s.Mnemonic,
		Key:        s.Key,
		PrivateKey: s.PrivateKey,
	}
}

func (c Identity) ToProto() *proto.Identity {
	return &proto.Identity{
		Mnemonic:   c.Mnemonic,
		Key:        c.Key,
		PrivateKey: c.PrivateKey,
	}
}

File: languages/js/src/entity/identity/identity.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";

export class Identity {
  mnemonic: string;
  key: string;
  privateKey: string;

  constructor(mnemonic: string, key: string, privateKey: string) {
    this.mnemonic = mnemonic;
    this.key = key;
    this.privateKey = privateKey;
  }

  public toProto(): identityEntitiesProto.Identity {
    return identityEntitiesProto.Identity.fromPartial({
      mnemonic: this.mnemonic,
      key: this.key,
      privateKey: this.privateKey
    });
  }

  static fromProto(r: identityEntitiesProto.Identity): Identity {
    return new Identity(r.mnemonic, r.key, r.privateKey);
  }
}
